sources/packages/backend/apps/queue-consumers/src/processors/schedulers/cas-integration/_tests_/cas-supplier-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,133 @@
+import { INestApplication } from "@nestjs/common";
+import { SystemUsersService } from "@sims/services";
+import {
+  createE2EDataSources,
+  E2EDataSources,
+  saveFakeStudent,
+} from "@sims/test-utils";
+import { QueueNames } from "@sims/utilities";
+import { CASSupplierIntegrationScheduler } from "../cas-supplier-integration.scheduler";
+import {
+  createTestingAppModule,
+  describeProcessorRootTest,
+  mockBullJob,
+} from "../../../../../test/helpers";
+import { CASSupplier, SupplierStatus } from "@sims/sims-db";
+import { CASService } from "../../../../services";
+import {
+  casLogonMockedResult,
+  getSupplierInfoFromCASMockedResult,
+} from "../../../../../test/helpers/mock-utils/cas-service.mock";
+
+describe(describeProcessorRootTest(QueueNames.CASSupplierIntegration), () => {
+  let app: INestApplication;
+  let processor: CASSupplierIntegrationScheduler;
+  let db: E2EDataSources;
+  let systemUsersService: SystemUsersService;
+  let casServiceMock: CASService;
+  const [supplierMockedResult] = getSupplierInfoFromCASMockedResult.items;
+
+  beforeAll(async () => {
+    const {
+      nestApplication,
+      dataSource,
+      casServiceMock: casServiceMockFromAppModule,
+    } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    casServiceMock = casServiceMockFromAppModule;
+    // Processor under test.
+    processor = app.get(CASSupplierIntegrationScheduler);
+    systemUsersService = app.get(SystemUsersService);
+  });
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+  });
+
+  it("Should finalize CAS supplier process with success when no supplier found.", async () => {
+    // Queued job.
+    const mockedJob = mockBullJob<void>();
+
+    // Act
+    const result = await processor.processCASSupplierInformation(mockedJob.job);
+
+    // Assert
+    expect(result).toStrictEqual([
+      "Process finalized with success.",
+      "Pending suppliers to update found: 0.",
+      "Records updated: 0.",
+    ]);
+    expect(
+      mockedJob.containLogMessages([
+        "Found 0 records to be updated.",
+        "CAS supplier integration executed.",
+      ]),
+    ).toBe(true);
+
+    expect(casServiceMock.casLogon).not.toHaveBeenCalled();
+    expect(casServiceMock.getSupplierInfoFromCAS).not.toHaveBeenCalled();
+  });
+
+  it("Should update CAS supplier table when found pending supplier information to be updated.", async () => {
+    const student = await saveFakeStudent(db.dataSource);
+
+    const casSupplier = new CASSupplier();
```

Comment:
Following the existing patterns we should create a factory (e.g. `createFakeCASSupplier`) to help with the entity creation and evolve from there.